tests/common/integration.rs

```diff
@@ -80,6 +80,21 @@ pub fn test_integration(package: impl AsRef<Path>, bindings: Option<String>) ->
 
         let python = target.get_venv_python(&venv_dir);
 
+        // Upgrade pip
```

Comment:
I'd like to avoid doing this in tests because this is slow and prone to break things in hard to understand ways. 

Is there a special reason you did this here and would it be possible to replace it by e.g. ensuring that virtualenv uses a sufficiently recent pip version and otherwise error while suggesting `virtualenv --upgrade-embed-wheels`?